lib/api/core/validation/types/geoShape.js

```diff
@@ -1,196 +1,223 @@
         result = false;
-      }
-      break;
-  }
+    }
 
-  if (isMulti) {
-    if (shape.coordinates.some(coordinate => !coordinateValidation(coordinate))) {
-      errorMessages.push(`One of the shapes in  the shape type "${shape.type}" has bad coordinates.`);
-      result = false;
+    if (isMulti) {
+      if (shape.coordinates.some(coordinate => !coordinateValidation(coordinate))) {
+        errorMessages.push(`One of the shapes in  the shape type "${shape.type}" has bad coordinates.`);
+        result = false;
+      }
+    }
+    else if (!coordinateValidation(shape.coordinates)) {
+      errorMessages.push(`The shape type "${shape.type}" has bad coordinates.`);
+      return false;
     }
-  }
-  else if (!coordinateValidation(shape.coordinates)) {
-    errorMessages.push(`The shape type "${shape.type}" has bad coordinates.`);
-    return false;
-  }
 
-  return result;
-};
+    return result;
+  }
 
-/**
- * @param {string[]} allowedShapes
- * @param {GeoShape} shape
- * @param {string[]} errorMessages
- * @returns {boolean}
- */
-GeoShapeType.prototype.checkStructure = function geoShapeCheckStructure (allowedShapes, shape, errorMessages) {
-  var result = true;
+  /**
+   * @param {string[]} allowedShapes
+   * @param {GeoShape} shape
+   * @param {string[]} errorMessages
+   * @returns {boolean}
+   */
+  checkStructure(allowedShapes, shape, errorMessages) {
+    let result = true;
+
+    if (!shape.type) {
+      errorMessages.push('The shape object has no type defined.');
+      return false;
+    }
 
-  if (!shape.type) {
-    errorMessages.push('The shape object has no type defined.');
-    return false;
-  }
+    if (!this.checkAllowedProperties(shape, allowedShapeProperties)) {
+      errorMessages.push('The shape object has a not allowed property.');
+      result = false;
+    }
 
-  if (!this.checkAllowedProperties(shape, allowedShapeProperties)) {
-    errorMessages.push('The shape object has a not allowed property.');
-    result = false;
-  }
+    if (allowedShapes.indexOf(shape.type) === -1) {
+      errorMessages.push('The provided shape type is not allowed.');
+      result = false;
+    }
 
-  if (allowedShapes.indexOf(shape.type) === -1) {
-    errorMessages.push('The provided shape type is not allowed.');
-    result = false;
-  }
+    if (shape.type === 'geometrycollection' && shape.coordinates) {
+      errorMessages.push('The coordinates property must not be provided for the "geometrycollection" shape type.');
+      result = false;
+    }
 
-  if (shape.type === 'geometrycollection' && shape.coordinates) {
-    errorMessages.push('The coordinates property must not be provided for the "geometrycollection" shape type.');
-    result = false;
-  }
+    if (shape.type !== 'geometrycollection' && (!shape.coordinates || !Array.isArray(shape.coordinates))) {
+      errorMessages.push(`The coordinates property must be provided for the "${shape.type}" shape type.`);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Split this 106 characters long line (which is greater than 80 authorized). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3ALineLength)